zarp/run/snakemake.py

```diff
@@ -1 +1,126 @@
 """Module for executing Snakemake workflows."""
+
+import subprocess
+from typing import List, Optional
+
+from zarp.config.models import (ExecModes, InitRun,
+                                DependencyEmbeddingStrategies)
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls.
+
+    Args:
+        run (InitRun): Run-specific parameters.
+
+    Attributes:
+        run_dict (dict): Dictionary with run-specific parameters.
+        success (bool): Indicator for successful snakemake run.
+        run_list (list[str]): List containing strings for snakemake call.
+
+    Example:
+        The example below expects a valid `Snakefile` (e.g. `touch Snakefile`)
+        in the current working directory.
+        It constructs a run with default values and runs it.
+        >>> mysnk = SnakemakeExecutor(InitRun())
+        >>> mysnk.prepare_run(snkfile = "Snakefile")
+        >>> mysnk.run()
+        >>> assert mysnk.get_success()
+
+    """
+
+    def __init__(self, run: InitRun) -> None:
+        """Class constructor."""
+        self.run_dict = run.dict()
```

Comment:
I'm not quite sure why you converted the model to a dictionary. It's much nicer to work with attributes: accessing them is shorter and easier to read and write compared to accessing dictionary values. Also, editors will be aware of them semantically, which can help prevent mistakes. It's a big part of the reason why we use models in the first place.

So I will change this to use the model instead.